src/openMVG/sfm/AlembicImporter.hpp

```diff
@@ -25,17 +20,14 @@ class AlembicImporter
 {
 public:
   explicit AlembicImporter(const std::string &filename);
-  ~AlembicImporter() = default;
+  ~AlembicImporter();
 
   void populate(sfm::SfM_Data &sfmdata, sfm::ESfM_Data flags_part = sfm::ESfM_Data::ALL);
 
 private:
-  bool readPointCloud(IObject iObj, M44d mat, sfm::SfM_Data &sfmdata, sfm::ESfM_Data flags_part);
-  bool readCamera(IObject iObj, M44d mat, sfm::SfM_Data &sfmdata, sfm::ESfM_Data flags_part, const index_t sampleFrame = 0);
   
-  //@todo complete the interface, also maybe parameters need to be passed by reference?
-  void visitObject(IObject iObj, M44d mat, sfm::SfM_Data &sfmdata, sfm::ESfM_Data flags_part = sfm::ESfM_Data::ALL);
-  IObject _rootEntity;
+  struct ObjImpl;
+  std::unique_ptr<ObjImpl> _objImpl;
```

Comment:
You can use `std::unique_ptr<Alembic::Abc::IObject>` with a forward declaration of IObject directly:
```
namespace Alembic {
namespace Abc {
class IObject;
}}
```
Or we can just rename it to `DataImpl` if the goal is to keep it symmetric with AlembicExporter.